client/my-sites/site-settings/form-seo.jsx

```diff
@@ -0,0 +1,379 @@
+		} );
+	},
+
+	submitSeoForm( event ) {
+		const { site, trackSubmission } = this.props;
+
+		if ( ! event.isDefaultPrevented() && event.nativeEvent ) {
+			event.preventDefault();
+		}
+
+		notices.clearNotices( 'notices' );
+
+		const filteredCodes = pickBy( this.state.verificationServicesCodes, isString );
+
+		const invalidCodes = [];
+		for ( const serviceName in filteredCodes ) {
+			if ( ! isValidVerificationCode( serviceName, filteredCodes[ serviceName ] ) ) {
+				invalidCodes.push( serviceName );
+			}
+		}
+
+		this.setState( { invalidCodes: invalidCodes } );
+		if ( invalidCodes.length > 0 ) {
+			notices.error( this.translate( 'Invalid site verification tag entered.' ) );
+			return;
+		}
+
+		this.setState( { isSubmittingForm: true } );
+
+		const updatedOptions = {
+			seo_meta_description: this.state.seoMetaDescription,
+			verification_services_codes: filteredCodes
+		};
+
+		site.saveSettings( updatedOptions, error => {
+			if ( error ) {
+				switch ( error.error ) {
+					case 'invalid_ip':
+						notices.error( this.translate( 'One of your IP Addresses was invalid. Please, try again.' ) );
+						break;
+					default:
+						notices.error( this.translate( 'There was a problem saving your changes. Please, try again.' ) );
+				}
+				this.setState( { isSubmittingForm: false } );
+			} else {
+				notices.success( this.translate( 'Settings saved!' ) );
+				this.markSaved();
+				this.setState( { isSubmittingForm: false } );
+
+				site.fetchSettings();
+			}
+		} );
+
+		trackSubmission();
+	},
+
+	render() {
+		const {
+			slug = '',
+			settings: {
+				blog_public = 1
+			} = {}
+		} = this.props.site;
+
+		const {
+			isSubmittingForm,
+			seoMetaDescription,
+			verificationServicesCodes,
+			showPasteError = false,
+			invalidCodes = []
+		} = this.state;
+
+		const isSitePrivate = parseInt( blog_public, 10 ) !== 1;
+		const isDisabled = isSitePrivate || isSubmittingForm;
+		const hasMetaError = seoMetaDescription && seoMetaDescription.length > 160;
+
+		const sitemapUrl = `https://${ slug }/sitemap.xml`;
+		const generalSettingsUrl = getGeneralSettingsUrl( slug );
+		const placeholderTagContent = '1234';
+
+		const googleTag = getMetaTagForService(
+			'google',
+			// The API returns 'false' for an empty array value, so we force it to an empty string if needed
+			get( verificationServicesCodes, 'google' ) || ''
+		);
+		const bingTag = getMetaTagForService(
+			'bing',
+			get( verificationServicesCodes, 'bing' ) || ''
+		);
+		const pinterestTag = getMetaTagForService(
+			'pinterest',
+			get( verificationServicesCodes, 'pinterest' ) || ''
+		);
+		const yandexTag = getMetaTagForService(
+			'yandex',
+			get( verificationServicesCodes, 'yandex' ) || ''
+		);
+
+		const isGoogleError = invalidCodes.indexOf( 'google' ) > -1;
+		const isBingError = invalidCodes.indexOf( 'bing' ) > -1;
+		const isPinterestError = invalidCodes.indexOf( 'pinterest' ) > -1;
+		const isYandexError = invalidCodes.indexOf( 'yandex' ) > -1;
```

Comment:
This had me giggling, because I wrote it like that but thought "@dmsnell wouldn't like it if I added the logic in render()", so I created the variables. But I agree in this case it looks better to do it there.